l1-contracts/src/core/libraries/Decoder.sol

```diff
@@ -108,6 +111,19 @@ library Decoder {
     bytes32 unencryptedLogsHashKernel2;
   }
 
+  // DECODING OFFSET CONSTANTS
+  // Where the start of trees metadata begins in the block
+  uint256 private constant START_TREES_BLOCK_HEADER_OFFSET = 0x80;
+
+  // Where the end of trees metadata begns in the block
+  uint256 private constant END_TREES_BLOCK_HEADER_OFFSET = 0x01c0;
+
+  // The size of the block header elements
+  uint256 private constant TREES_HEADER_SIZE = 0x140;
+
+  // Where the metadata ends and the block data begins.
+  uint256 private constant METADATA_OFFSET = 0x0300;
```

Comment:
All the other things was named with some "sub-header", seems like it would make sense to use header here instead of metadata. Would also match better with some of the comments we have further down.